modules/stac-server/lambda/waiting_for_opensearch/main.py

```diff
@@ -16,17 +16,22 @@ def lambda_handler(event, context):
 def wait_for_opensearch_serverless_collection(collection):
     collection_active = False
     while not collection_active:
-        logger.info(f'Checking if opensearch serverless collection {collection} is active')
-        collection_list = opensearchserverless.list_collections(
-            collectionFilters={
-                'name': COLLECTION_NAME
-            }
-        )
-        if 'collectionSummaries' in collection_list and len(collection_list['collectionSummaries']) > 0 and collection_list['collectionSummaries'][0]['status'] == 'ACTIVE':
-            logger.info(f'Found active opensearch serverless collection {collection_list}')
-            collection_active = True
-        elif 'collectionSummaries' in collection_list and len(collection_list['collectionSummaries']) > 0 and collection_list['collectionSummaries'][0]['status'] == 'FAILED':
-            raise Exception(f'Opensearch serverless collection {collection_list} has failed to create...')
-        else:
-            logger.info(f'Waiting for opensearch serverless collection {collection} to be available, sleeping...')
-            time.sleep(5)
+            logger.info(f'Checking if opensearch serverless collection {collection} is active')
+            collection_list = {}
+            try:
+                collection_list = opensearchserverless.list_collections(
+                    collectionFilters={
+                        'name': COLLECTION_NAME
+                    }
+                )
+            except Exception as e:
+                logger.info(f'Permissions for {collection} or OpenSearch Serverless may not be available yet, sleeping...: {e}')
+                collection_list = {}
+            if 'collectionSummaries' in collection_list and len(collection_list['collectionSummaries']) > 0 and collection_list['collectionSummaries'][0]['status'] == 'ACTIVE':
+                logger.info(f'Found active opensearch serverless collection {collection_list}')
+                collection_active = True
+            elif 'collectionSummaries' in collection_list and len(collection_list['collectionSummaries']) > 0 and collection_list['collectionSummaries'][0]['status'] == 'FAILED':
+                raise Exception(f'Opensearch serverless collection {collection_list} has failed to create...')
+            else:
+                logger.info(f'Waiting for opensearch serverless collection {collection} to be available, sleeping...')
+                time.sleep(5)
```

Comment:
This looks like it's indented eight instead of four. The fact that CI didn't catch this is a problem. We should have (at minimum) formatting and linting set up for Python code, and preferably some sort of smoke test as well.

Reply:
fixed